Connected System Template (CST) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/helloworld/templates/GoogleDriveSendFileIntegrationTemplate.java

```diff
@@ -51,78 +57,44 @@ protected IntegrationResponse execute(
     Drive drive = new Drive.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance(), credential).build();
-    IntegrationDesignerDiagnostic.IntegrationDesignerDiagnosticBuilder diagnosticBuilder =  IntegrationDesignerDiagnostic.builder();
+    IntegrationDesignerDiagnostic.IntegrationDesignerDiagnosticBuilder diagnosticBuilder = IntegrationDesignerDiagnostic
+        .builder();
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
 
-    final long start = System.currentTimeMillis();
-    long end;
+    Map<String, String> responseDiagnostics = new HashMap<>();
+    Map<String, Object> result = new HashMap<>();
     try {
       File responseFile = drive
           .files()
           .create(file, mediaContent)
           .setFields("name, id, mimeType")
           .execute();
-      result.put("Name", responseFile.getName());
-      result.put("File ID", responseFile.getId());
-      result.put("MIME Type", responseFile.getId());
-
-      responseDiagnostic.put("Name", responseFile.getName());
-      responseDiagnostic.put("File ID", responseFile.getId());
-      responseDiagnostic.put("MIME Type", responseFile.getMimeType());
-      end = System.currentTimeMillis();
+      responseDiagnostics = sendFileDiagnosticsFactory.getResponseDiagnostics(responseFile);
+      result = sendFileDiagnosticsFactory.getResult(responseFile);
     } catch (GoogleJsonResponseException e) {
-      if (e.getStatusCode() == 401) {
-        //Google returns a 401 exception if your credential is not authorized or expired. Throw a
-        //ExpiredTokenException when this happens, Appian will try to refresh the token for up to
-        //3 times for you.
-        throw new ExpiredTokenException();
-      }
-      end = System.currentTimeMillis();
-      IntegrationDesignerDiagnostic diagnostic = diagnosticBuilder.addExecutionTimeDiagnostic(end - start)
-          .addRequestDiagnostic(requestDiagnostic).build();
-
-      IntegrationError error =  IntegrationError.builder()
-          .title("Google Exception Status: " + e.getStatusMessage())
-          .message(e.getMessage())
-          .build();
-
-      return IntegrationResponse
-          .forError(error)
-          .withDiagnostic(diagnostic)
-          .build();
+      Map<String,Object> requestDiagnostics = sendFileDiagnosticsFactory.getRequestDiagnostics(document,
+          connectedSystemConfiguration, integrationConfiguration);
+      return GoogleJsonResponseExceptionHandler.handleException(e, diagnosticBuilder,
+          requestDiagnostics, stopwatch);
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
 
+    long elapsed = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+    Map<String,Object> requestDiagnostics = sendFileDiagnosticsFactory.getRequestDiagnostics(document,
+        connectedSystemConfiguration, integrationConfiguration);
     IntegrationDesignerDiagnostic diagnostic = diagnosticBuilder
```

Comment:
Same comment as before. I think the diagnostics factory should make diagnostics.